aws-datastore/src/main/java/com/amplifyframework/datastore/DataStoreConfiguration.java

```diff
@@ -0,0 +1,340 @@
+    public int getSyncMaxRecords() {
+        return this.syncMaxRecords;
+    }
+
+    /**
+     * Gets the number of items that should be requested in page, from AppSync, during
+     * a sync operation.
+     * @return Desired size of a page of results from an AppSync sync response
+     */
+    @IntRange(from = 0)
+    public int getSyncPageSize() {
+        return this.syncPageSize;
+    }
+
+    /**
+     * Builds instances of {@link AWSDataStorePlugin} by providing a variety of
+     * configuration methods.
+     */
+    public static final class Builder {
+        private DataStoreErrorHandler dataStoreErrorHandler;
+        private DataStoreConflictHandler dataStoreConflictHandler;
+        private Long syncIntervalMs;
+        private Integer syncMaxRecords;
+        private Integer syncPageSize;
+
+        private Builder() {
+            this.dataStoreErrorHandler = DefaultDataStoreErrorHandler.instance();
+            this.dataStoreConflictHandler = ApplyRemoteConflictHandler.instance(dataStoreErrorHandler);
+            this.syncIntervalMs = DEFAULT_SYNC_INTERVAL_MINUTES;
+            this.syncMaxRecords = DEFAULT_SYNC_MAX_RECORDS;
+            this.syncPageSize = DEFAULT_SYNC_PAGE_SIZE;
+        }
+
+        /**
+         * A handler that will be invoked whenever there is a conflict between two model instances,
+         * one in the local store, and one from the remote server, as received from a sync operation.
+         * @param dataStoreConflictHandler A handler to invoke upon sync conflicts
+         * @return Current builder
+         */
+        @NonNull
+        public Builder dataStoreConflictHandler(@NonNull DataStoreConflictHandler dataStoreConflictHandler) {
+            this.dataStoreConflictHandler = Objects.requireNonNull(dataStoreConflictHandler);
+            return Builder.this;
+        }
+
+        /**
+         * Sets a handler function to be applied when the DataStore encounters an unrecoverable error
+         * in one of its ongoing background operations (model synchronization).
+         * @param dataStoreErrorHandler A handler for unrecoverable background errors
+         * @return Current builder instance
+         */
+        @NonNull
+        public Builder dataStoreErrorHandler(@NonNull DataStoreErrorHandler dataStoreErrorHandler) {
+            this.dataStoreErrorHandler = Objects.requireNonNull(dataStoreErrorHandler);
+            return Builder.this;
+        }
+
+        /**
+         * Sets the duration of time after which delta syncs will not be preferred over base syncs.
+         * @param syncIntervalMs The amount of time that must elapse for delta syncs to not be considered
+         * @return Current builder instance
+         */
+        @NonNull
+        public Builder syncIntervalMs(@IntRange(from = 0) long syncIntervalMs) {
+            //Only set this value if the incoming value is not equal to the default
+            if (syncIntervalMs != DEFAULT_SYNC_INTERVAL_MINUTES) {
```

Comment:
Uh oh! This builder didn't get updated. This still says `ms`, but it should say:

```
@NonNull
public Builder syncIntervalInMinutes(@IntRange(from = 0) long syncIntervalInMinutes) {
    this.syncIntervalMs = TimeUnit.MINUTES.toMilliseconds(syncIntervalInMunutes);
    return Builder.this;
}
```

No `if` needed.

As per comments further down -- this is a ~good place to do the unit conversion into `ms`, I think.

(Human beings like minutes, but computers like milliseconds.)

Reply:
Okay cool I understand the intent now! Thanks

Reply:
My latest commit makes the scalar values nullable and returns defaults from the getters if the underlying field is null